### DISCUSSION TITLE
Sort related records

### DISCUSSION THREAD

**Discussion by vinyll** [pocketbase/pocketbase#4522] [Q&A]
Hey there!

I moved from another BaaS to PocketBase and I'm very impressed how fully functional and reliable it is 🚀 

I'm now into details as I'm trying to sort a relation that is expanded by created date descending (latest on top of my results) using the API.

I see there is a `{ sort: "-created" }` but that's for the collection queried, not for expanded relation.
Is there any mean to do so out-of-the-box?

:)

**Comment by ganigeorgiev** [comment]
The question is too vague and it is not clear what do you mean by "to sort a relation that is expanded"? Is it a single or multiple `relation` that is being expanded?

In general pagination, sorting, and filtering works only for the collection that is queried.

You can use dot-notation (eg. `yourRelField.someOtherField`) to target the relation fields but that will sort its owner record that contains the `expand`, not the expanded value.

**Reply by devbyVeriTech** [reply]
The issue is, I was able to sort the collection record.
Now I have a related record which I want to sort in descending order but I can’t

**Comment by vinyll** [comment]
Right, here's a simplified example.
A collection of articles has related comments `1->n` that have a `created` date column.
I would need to retrieve articles along with their comments where comments would be sorted by `created DESC`.
In my case I call the `pb.collection('articles').getOne('RECORD_ID', { expand: "comments" }`

Maybe something like `{ sort: "-comments.created" }` or something could make some sense?

**Reply by devbyVeriTech** [reply]
Were you able to figure it out?

**Reply by vinyll** [reply]
I have not.
In the meantime I sort the data in the frontend after retreival.

**Reply by devbyVeriTech** [reply]
Can you share your method?

**Reply by vinyll** [reply]
I just retrieve the collection and then I run `collection.sort((a, b) => return a.created < b.created ? -1 : 1)` to sort in JS.

The lib I use for the frontend is [Lego](https://lego.js.org/v2/) si it might be too specific to get further in details.

**Reply by esaesa** [reply]
> Right, here's a simplified example. A collection of articles has related comments `1->n` that have a `created` date column. I would need to retrieve articles along with their comments where comments would be sorted by `created DESC`. In my case I call the `pb.collection('articles').getOne('RECORD_ID', { expand: "comments" }`
> 
> Maybe something like `{ sort: "-comments.created" }` or something could make some sense?

This is good for non paginated data otherwise it will be misleading
